src/server/table.h

```diff
@@ -79,7 +79,47 @@ struct DbTableStats {
   DbTableStats& operator+=(const DbTableStats& o);
 };
 
-using LockTable = absl::flat_hash_map<std::string, IntentLock>;
+// Table for recording locks that uses string_views where possible. LockTable falls back to
+// strings for locks that are used by multiple transactions. Keys used with the lock table
+// should be normalized with GetLockKey
+class LockTable {
+ public:
+  size_t Size() const;
+  std::optional<const IntentLock> Find(std::string_view key) const;
+
+  bool Acquire(std::string_view key, IntentLock::Mode mode);
+  void Release(std::string_view key, IntentLock::Mode mode);
+
+  auto begin() const {
+    return locks_.cbegin();
+  }
+
+  auto end() const {
+    return locks_.cbegin();
+  }
```

Comment:
Actually interesting how this leaks a private type but the compiler doesn't complain 🤷🏻‍♂️

Reply:
apparently it's a language feature